augur/util_support/color_parser.py

```diff
@@ -0,0 +1,31 @@
+from io import TextIOWrapper
+import functools
+from pkg_resources import resource_stream
+
+from augur.util_support.color_parser_line import ColorParserLine
+
+
+class ColorParser:
+    def __init__(self, *, mapping_filename, use_defaults=True):
+        self.mapping_filename = mapping_filename
+        self.use_defaults = use_defaults
+
+    @property
+    @functools.lru_cache()
+    def mapping(self):
+        colors = {}
+
+        if self.use_defaults:
+            with resource_stream("augur", "data/colors.tsv") as stream:
+                with TextIOWrapper(stream, "utf-8") as defaults:
+                    colors = {**colors, **self.parse_file(defaults)}
+
+        if self.mapping_filename:
+            with open(self.mapping_filename, encoding="utf-8") as mapping:
+                colors = {**colors, **self.parse_file(mapping)}
+
+        return colors
```

Comment:
The original `read_colors` function [returns a dictionary of lists indexed by trait and referring to trait value and hex code pairs](https://github.com/nextstrain/augur/pull/588/files#diff-b5772e65b5a8175ae1a542a1c7ad0e38L414-L418). The new class returns a dictionary of hex codes indexed by trait and trait value.

The following change resolves the issue for me locally:

```diff
diff --git a/augur/util_support/color_parser.py b/augur/util_support/color_parser.py
index b44f170..af23300 100644
--- a/augur/util_support/color_parser.py
+++ b/augur/util_support/color_parser.py
@@ -1,3 +1,4 @@
+from collections import defaultdict
 from io import TextIOWrapper
 import functools
 from pkg_resources import resource_stream
@@ -24,7 +25,11 @@ class ColorParser:
             with open(self.mapping_filename, encoding="utf-8") as mapping:
                 colors = {**colors, **self.parse_file(mapping)}

-        return colors
+        color_map = defaultdict(list)
+        for (trait, trait_value), hex_code in colors.items():
+            color_map[trait].append((trait_value, hex_code))
+
+        return color_map

     def parse_file(self, file):
         pairs = [ColorParserLine(line).pair() for line in file]
```

Reply:
Ah, of course. A silly mistake on my part. Updated.